tests/queries/0_stateless/02983_variant_data_types.sh

```diff
@@ -0,0 +1,312 @@
+    run_select "Int32"
+    drop_table
+    
+    create_table "$1" "Int64"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int64) from numbers(3);"
+    run_select "Int64"
+    drop_table
+    
+    create_table "$1" "Int128"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int128) from numbers(3);"
+    run_select "Int128"
+    drop_table
+    
+    create_table "$1" "Int256"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int256) from numbers(3);"
+    run_select "Int256"
+    drop_table
+}
+
+function test_floating_point_numbers()
+{
+    echo "Floating Point Types"
+
+    create_table "$1" "Float32"
+    $CH_CLIENT -q  "insert into test select 1, round(1 - 0.9, 4)::Float32 FROM system.numbers LIMIT 3;"
+    run_select "Float32"
+    drop_table
+
+    create_table "$1" "Float64"
+    $CH_CLIENT -q  "insert into test select 1, (1 - 0.9)::Float64 FROM system.numbers LIMIT 3;"
+    run_select "Float64"
+    drop_table
+    
+    #create_table "$1" "Decimal32"
+    #$CH_CLIENT -q  "insert into test select 1, toDecimal32(round(randCanonical(), 3)::Decimal32 , 3) FROM system.numbers LIMIT 3;"
+    #run_select "Decimal32"
+    #drop_table
+}
+
+function test_boolean()
+{
+    echo "Boolean Types"
+
+    create_table "$1" "Bool"   # create table works for Boolean but select only works for .Bool
+    $CH_CLIENT -q  "insert into test values (1, true);"
+    run_select "Bool"
+    drop_table
+}
+
+function test_strings()
+{
+    echo "String Types"
+
+    create_table "$1" "String"
+    $CH_CLIENT -q  "insert into test values (1, 'Hello');"
+    run_select "String"
+    drop_table
+
+    create_table "$1" "FixedString(5)"
+    #$CH_CLIENT -q  "insert into test values (1, '12345');"
+    #run_select "FixedString(5)"  # Cannot select referring to data type FixedString or FixedString(5)
+    drop_table
+}
+
+function test_dates()
+{
+    echo "Date Types"
+
+    create_table "$1" "Date"
+    $CH_CLIENT -q  "insert into test values (1, '2019-01-01');"
+    run_select "Date"
+    drop_table
+
+    create_table "$1" "Date32"
+    $CH_CLIENT -q  "insert into test values (1, '1969-01-01');"
+    run_select "Date32"
+    drop_table
+
+    create_table "$1" "DateTime"
+    $CH_CLIENT -q  "insert into test values (1, '2019-01-01 00:00:00');"
+    run_select "DateTime"
+    drop_table
+
+    create_table "$1" "DateTime64"
+    $CH_CLIENT -q  "insert into test values (1, '2299-01-01 00:00:00');"
+    # run_select "DateTime64" # Cannot select referring to data type DateTime64 or DateTime64(3)
+    drop_table
+}
+
+function test_json()
+{
+    echo "JSON Types"
+
+    #create_table "$1" "JSON" # Variant accepts the JSON type with exception : "Code: 36. DB::Exception: Positional options are not supported. (BAD_ARGUMENTS)"
```

Comment:
JSON type is experimental and current implementation will be removed (new one is described [here](https://github.com/ClickHouse/ClickHouse/issues/54864) and will use `Variant` inside), so it doesn't make sense to test `Variant` with JSON right now (it's expected to not work)

Reply:
Yes it makes sense, I was trying unsupported types also as a negetive cases to make sure proper errors are there.